baseapp/baseapp.go

```diff
@@ -848,13 +853,65 @@ func createEvents(msg sdk.Msg) sdk.Events {
 	return sdk.Events{msgEvent}
 }
 
+// DefaultPrepareProposal returns the default implementation for processing an ABCI proposal.  The application's
+// mempool is enumerated and all valid transactions are added to the proposal.  Transactions are valid if they:
+//
+// 1) Successfully encode to bytes.
+// 2) Are valid (i.e. pass runTx, AnteHandler only)
+//
+// Enumeration is halted once RequestPrepareProposal.MaxBytes of transactions is reached or the mempool is exhausted.
+// Note that step (2) is identical to the validation step performed in DefaultProcessProposal.  It is very important
+// that the same validation logic is used in both steps, and applications must ensure that this is the case in
+// non-default handlers.
+func (app *BaseApp) DefaultPrepareProposal() sdk.PrepareProposalHandler {
+	return func(ctx sdk.Context, req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
```

Comment:
curious why this can't return an error instead of using `panic`?

Reply:
It could but there is no error channel back to TM, so baseapp.PrepareProposal can only panic also.